### ISSUE TITLE
Large repo size (200 MB)

### ISSUE THREAD

**Issue by BenediktBurger** [PyMoDAQ/pymodaq_data#7]
This repository is quite large with over 200 MB for the few content.
I guess it contains icons etc. from pymodaq due to its common history with the full PyMoDAQ package.
Removing the old part of the history could reduce the repository size.

**Comment by seb5g** [comment]
indeed do you know how to do it?

**Comment by seb5g** [comment]
we could do that:
https://stackoverflow.com/questions/4515580/how-do-i-remove-the-old-history-from-a-git-repository

but then which depth to use... I tried with only 1 and it's still 37 Mb. What would be "acceptable" as a repo size?

(I also removed the documentation as it is only in the pymodaq repo at the moment)

**Comment by BenediktBurger** [comment]
37 MB is better than 200 MB, reducing the size by 80%.

I stumbled upon the size due to noticeable time cloning the repository and thought it might be good to improve cloning time (and space consumption) by focusing on the necessary parts.

**Comment by malik-irain** [comment]
I tried to look at it, especially because it concerns pymodaq_utils and pymodaq_gui too. There are no free/easy solutions:

The repos are big for two reasons:
1. They contain the full history before the split from PyMoDAQ.
2. They track binary files (pdf, images, data, etc...) that were removed from the repo but are still kept in the history, hence sizing up the repo.

The solution in  @seb5g's link  with `git clone <repo> **--depth=n**` just tell you to download the last **n** commits, so only partially fixing the problem (the full repo is still big).  It is then necessary to rewrite commit history to remove what's not necessary.

I tried  to "reset" this repo at the commit of the first tag/release locally on my computer. To keep it short, it regenerated the commits with new hashes, so I had to delete the tags (they can still be regenerated on the new commit hashes). I ended up with a size of 40.2Mb for the full history. As hashes changed, if theses changes are pushed, it invalidates the existing forks.

It still contains some unused binary files mainly from the doc, I guess it's also possible to modify the remaining commits to resolve it but this would take some time to modify commit history to remove them.

Then, another solution would be to recreate a fresh repository, it gives a size of 1,1Mb but we lose history and forks are also invalidated.

So, to sum up, there's two solutions to reduce the size (they don't affect the already released versions on pypi):
1. The hard way, rewrite commit history (~40Mb),  version tags have to be regenerated with the new commits hash, and forks will (probably) be invalidated.
2. The easy way, recreate a fresh repository (~1Mb), we would lose some history and forks will also be invalidated.

@rgeneaux  @Ashwolaa  An opinion on this?

**Comment by Ashwolaa** [comment]
I vote for the fresh repository.
Now is the best time for a fresh start, these new repositories are now detaching from the base use of pymodaq so there is no real need to keep track of the history for me. The latter can still be found (if needed) in the pymodaq's folder?

Does that mean our forks / PR will be aborted ?

**Comment by malik-irain** [comment]
What's merged is already merged so no problem with that.

As for the rest, it will only invalidate/break the link between your repository and PyMoDAQ/pymodaq_data (utils and gui too), but the repo will not disappear.

I will do some more testing on my side to ensure the smoothest transition.

**Comment by seb5g** [comment]
For sure, we should finish the started PR before proceeding further. There is none on pymodaq_data but at least 2 on pymodaq_gui I didn't find time to properly test... So we can first test on pymodaq_utils and data and make sure everything is still fine, then do gui last when the PR are merged

**Comment by rgeneaux** [comment]
I also vote for the fresh repository! There aren't many forks yet and everything up to <5.0.0 is saved in the PyMoDAQ/pymodaq repository, this seems like a good time to do it.

**Comment by BenediktBurger** [comment]
Fresh repository!
